### ISSUE TITLE
Allow selecting font size 

### ISSUE THREAD

**Issue by rnc** [keepassxreboot/keepassxc#6822]
## Summary
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Provide a brief overview of what the new feature is all about )

While I *really* like the new monospace font in notes I it would be really useful if I could customise the font size (_cough_ for my old eyes :-) )

## Examples
[NOTE]: # ( Show us a picture or mock-up of your proposal )

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )

**Comment by droidmonkey** [comment]
We take the font size for your system, so it is the same size as all your other fonts. Adjust your system font size. Note, this may not work if you are using Linux and the snap build.

**Comment by matthudsonatx** [comment]
I'm looking forward to this feature. Gnome's font size control has 2 settings and is entirely insufficient for accessibility purposes.

**Comment by seperman** [comment]
After upgrading to 2.7.0 suddenly the font size is way smaller. So there was a regression introduced. I'm on Ubuntu 20.0.4 LTS. I have not touched my system font size in many months. I might downgrade the KeepPassXC version just so I can read the stuff on the 4k monitor.

**Comment by droidmonkey** [comment]
I assume you are using the snap. Welcome to the wonderful world of inconsistent Linux UX! You can use the Qt env vars to control your scaling: https://keepassxc.org/docs/KeePassXC_UserGuide.html#_command_line_options

QT_SCREEN_SCALE_FACTORS

**Comment by seperman** [comment]
@droidmonkey 
Thanks for the prompt response. Yes I'm using Snap. Ok sounds like I should always launch it from the terminal instead of the app bar.

**Comment by michaelk83** [comment]
> sounds like I should always launch it from the terminal instead of the app bar.

You can edit the app's `.desktop` file. https://askubuntu.com/a/144971/1193612
Or maybe right-click the link -> Edit, or something along those lines.

**Comment by seperman** [comment]
Thanks for the info @michaelk83 
It seems like when installing via snap, it doesn't create a `.dekstop` file. I wonder if I should stop using Keepass via snap.

**Comment by sliedes** [comment]
Where does the "system font size" come from? I'm on Ubuntu, running regolith (essentially Gnome). After compiling keepassxc from source, the UI is fairly big. I tried the somewhere-offered strategy of configuring qt5 with qt5ct. Changing font sizes there _does_ affect the font sizes in keepassxc, but still they are much larger in KeepassXC (started with QT_QPA_PLATFORMTHEME=qt5ct ./keepassxc) than in qt5ct (started with QT_QPA_PLATFORMTHEME=qt5ct qt5ct).

Here's a picture of qt5ct and keepassxc running side by side, started from the same terminal with the exact same environment variables:

![image](https://user-images.githubusercontent.com/2307997/227945321-63a6f57d-4d00-4c94-a0f9-f722022619e9.png)

**Comment by phoerious** [comment]
Looks like you are using the AppImage, which doesn't use the system theme. Install the app via the PPA to get full integration.

**Comment by sliedes** [comment]
At least I was just building it from source and running it directly from the build/src directory. I assume that should use the system theme?

**Comment by phoerious** [comment]
If you have one configured, yes.

**Comment by GitHubinatrix** [comment]
> We take the font size for your system, so it is the same size as all your other fonts. Adjust your system font size. Note, this may not work if you are using Linux and the snap build.

Apparently it also does not work sufficiently on Windows 10 if users happens to use its

_**Settings > System > Display > Scale and layouts > Change the size of text, apps, and other items > 150%**_

Almost every program succumbs to this and acts very accordingly- i.e. makes all of its texts bigger. Unfortunately KeePassXC is somewhere between the category of user-friendly software for people with impaired vision and the group of programs suitable for the ant-people community

In other words: I would like to be able to control the size of elements - most importantly text in boxes with passwords

I have a 38" monitor with 21:9 aspect ratio [set to 150%] - and yet still have issues with readability in KeePassXC

**Comment by droidmonkey** [comment]
You can use the Qt environment variable: QT_SCALE_FACTOR and set it to 1.5. Sometimes Qt doesn't properly pickup the desktop scaling. I am planning to introduce font choice in 2.8.0

**Comment by VA1DER** [comment]
I don't necessarily want the font size in an app to match the system font size, which controls the OS's font UI font size.

I also would very much like to not just change the fonts sizes in KeepassXC but also change the font themselves.  For example, the font used to display the password should always use a monospace font that distinguishes between O and 0, I and l and 1 clearly.  Yes, I'm aware of the "exclude lookalike characters".  It's better to display them clearly than to eliminate them.

A font selection would be a great benefit.

**Comment by JohnLGalt** [comment]
I have to agree. Adjusting font sizes in Windows UI is not all that good to begin with. UIs vary, and where it looks all right in one place it looks too large in others when enlarged, and vice versa if left at default 100% scaling.

I would much prefer granular control over **both** the size and the actual fonts used. I have an array of fonts available to me at any given time, and force usage of fonts for other apps, even my browser in some cases, completely independently of the OS.

Hell, I even change **default** fonts on Terminal, CMD and Notepad. Having 52 year old eyes coupled with a 4K monitor kinda forces me to.

**Comment by GitHubinatrix** [comment]
Yes, adjusting OS's font is a straight way to Messy Town

And that is why you set Windows 10 to

**_Settings > Ease Of Access > Display > Make everything bigger > Change the size of apps and text on the main display > 150%_**

and bigger fonts in text editors and also zoom in the view in Excel and such

**Comment by vkuznet** [comment]
I also beg developers to raise priority for this feature request. With aging population it will become more and more requested. Even though on this ticket people asked about Windows and Ubuntu I think all platforms need this. I'm on macOS and I want to request this feature on macOS too.

Meanwhile I found solution for macOS. Hope it may help others. You need to open your app with `QT_SCALE_FACTOR` environment, e.g.
```
QT_SCALE_FACTOR=1.5 open /Applications/KeePassXC.app
```
If you need to make it system wide, go to your default shell, e.g.`~/.bashrc` and add `QT_SCALE_FACTOR=1.5`, then it will be presented in your environment when you log into your account and will be used by other apps. Not ideal, but those who know how to tweak shell it is a way to go (I would assume it will work this way on any Linux OSes too). But it would be nice to have it as part of app settings itself that those folks who have less experience with terminal and shell can easily use it too.
